.azure-pipelines/ultimate-pipeline.yml

```diff
@@ -261,6 +261,36 @@ stages:
       displayName: Upload working directory after the build
       artifact: build-macos-working-directory
 
+- stage: check_changes
```

Comment:
This feels like a _very_ useful job (which _should_ be built into AzDo IMO). I think it would be nice to extract this for lots of our other jobs/stages too.

Just a note to self, I think keeping it separate for now is fine, but it might be worth moving to a separate step or something later once we spec it all out

Reply:
Well, they have `trigger` for pipelines :) 
I can refactor this to nuke target, and make it more generic.

Reply:
Moved this to nuke, so we can use it other steps as well.
If you already know filter conditions, I can add them to current PR.